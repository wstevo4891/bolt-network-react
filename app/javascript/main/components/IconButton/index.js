// Icon Button Component

import React from 'react'
import PropTypes from 'prop-types'

import makeToggleable from '@helpers/makeToggleable'

import Button from '../Button'
import Icon from '../Icon'

function buildIconProps(icon, iconProps) {
  if (icon) {
    return Object.assign({}, iconProps, { icon })
  }

  return iconProps
}

function buildButtonProps(buttonProps, handleClick) {
  if (buttonProps.handleClick) return buttonProps

  return Object.assign({}, buttonProps, { handleClick })
}

const IconButton = (props) => {
  const propsForButton = buildButtonProps(props.buttonProps, props.handleClick)
  const propsForIcon = buildIconProps(props.icon, props.iconProps)

  return (
    <Button {...propsForButton}>
      {props.textPlacement === 'left' && props.text}
      <Icon {...propsForIcon} />
      {props.textPlacement === 'right' && props.text}
    </Button>
  ) 
}

IconButton.propTypes = {
  buttonProps: PropTypes.shape(Button.propTypes),
  handleClick: PropTypes.func,
  icon: PropTypes.string,
  iconProps: PropTypes.shape(Icon.propTypes),
  text: PropTypes.string,
  textPlacement: PropTypes.oneOf(['left', 'right']),
}

IconButton.defaultProps = {
  buttonProps: {},
  handleClick: () => void {},
  icon: null,
  iconProps: null,
  text: null,
  textPlacement: 'right',
}

const ToggleIconButton = makeToggleable(IconButton)

export {
  IconButton,
  ToggleIconButton
}
