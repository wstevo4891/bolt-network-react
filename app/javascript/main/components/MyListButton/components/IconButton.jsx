// MyList IconButton

import React from 'react'

const IconButton = (props) => (
  <li className="poster-btn poster-btn-my-list">
    <button onClick={props.handleClick}>
      <i className={props.iconClass}></i>
    </button>
  </li>
)

export default IconButton
