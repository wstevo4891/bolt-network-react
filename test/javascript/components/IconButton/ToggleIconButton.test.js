import React from 'react'
import { shallow } from 'enzyme'

import { ToggleIconButton } from '@components/IconButton'

describe('ToggleIconButton', () => {
  const shallowComponent = () => shallow(
    <ToggleIconButton
      status={true}
      iconOptions={['fa-volume-off', 'fa-volume-up']}
    />
  )

  let component = shallowComponent()

  it('should render correctly in "debug" mode', () => {
    expect(component).toMatchSnapshot()
  })

  describe('toggle function', () => {
    beforeEach(() => {
      component = shallowComponent()
    })

    it('should render with expected state', () => {
      expect(component.state().status).toEqual(true)
    })
  
    it('should toggle state.status', () => {
      component.dive().simulate('click')
  
      expect(component.state().status).toEqual(false)
    })
  
    it('should render with expected props', () => {
      expect(component.find('IconButton').props().icon).toEqual('fa-volume-up')
    })
  
    it('should toggle icon prop', () => {
      component.dive().simulate('click')
  
      expect(component.find('IconButton').props().icon).toEqual('fa-volume-off')
    })
  })
})
