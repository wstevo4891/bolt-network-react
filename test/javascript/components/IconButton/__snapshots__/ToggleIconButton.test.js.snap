// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleIconButton should render correctly in "debug" mode 1`] = `
<IconButton
  buttonClass=""
  handleClick={[Function]}
  icon="fa-volume-up"
  text=""
/>
`;
